### PR TITLE
Allow dependabot to check github actions weekly

### PR DESCRIPTION
**Pull request by jlosito**
Allow dependabot to check github actions weekly

https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/keeping-your-actions-up-to-date-with-dependabot

### CODE REVIEW

**Comment by JukkaL** [comment]
Can you summarize the benefit of this? I quickly browsed the docs but I'm still not quite sure what the concrete impact would be.

**Comment by jlosito** [comment]
This change would upgrade the version of the follow action this project uses.

https://github.com/python/mypy/blob/master/.github/workflows/test.yml#L40

It looks like the most recent version of this action is v2 https://github.com/marketplace/actions/setup-python

As new versions of this action comes out, PRs will be submitted automatically to keep it up to date as well as any other github actions which this project uses.
